design/automate_release_design.md

```diff
@@ -0,0 +1,204 @@
+# Solution Design - Secrets Provider Release
+
+## Table of Contents
+
+[//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
+
+## Glossary
+
+[//]: # "Describe terms that will be used throughout the design"
+[//]: # "You can use this tool to generate a table - https://www.tablesgenerator.com/markdown_tables#"
+
+| **Term**                                                     | **Description**                                              |
+| ------------------------------------------------------------ | ------------------------------------------------------------ |
+| [Github actions](https://docs.github.com/en/free-pro-team@latest/actions) | CI/CD platform that is event-driven to help automate task and execute workflows in repositories |
+
+## Useful links
+
+[//]: # "Add links that may be useful for the reader"
+
+Release documentation
+
+- [Secrets Provider release documentation](https://github.com/cyberark/secrets-provider-for-k8s/blob/master/CONTRIBUTING.md#releases)
+
+Open issues:
+
+- [Secrets Provider - Release automation](https://app.zenhub.com/workspaces/palmtree-5d99d900491c060001c85cba/issues/cyberark/secrets-provider-for-k8s/233)
+
+- [Secrets Provider - Github action for dispatching events](https://github.com/cyberark/secrets-provider-for-k8s/issues/239)
+
+- [Community - Guidelines and tools for releasing artifacts to separate suite components](https://github.com/cyberark/community/issues/77)
+
+PR:
+
+- https://github.com/cyberark/secrets-provider-for-k8s/pulls
+
+## Background
+
+[//]: # "Give relevant background for the designed feature. What is the motivation for this solution?"
+
+The Secrets Provider release process has become a long and heavily manual process. This has become more evident with the introduction of Helm to the project which now adds additional steps to the release process. 
+
+Goals for this effort:
+
+1. Automate the parts we can.
+2. Knowledge sharing by providing guidelines for event-driven actions across multiple repositories.
+3. Push tag on every master build for easier testing and to better align ourselves with other open source projects
```

Comment:
it may be described in the rest of the doc but in case not - do we want also to add a tar with the image to the release? I created [this issue](https://github.com/cyberark/conjur-authn-k8s-client/issues/168) for the authn-client which we can do for this repo as well

Reply:
also - you don't need to actually list the numbers in `md` files. instead, you can just use `1` and the UI will show numbers. It can help for adding new entries in the middle without the need to change the numbers

Reply:
@orenbm This is a neat idea thanks for suggesting! Why do we add a tar for the image in a release btw? I don't remember doing this in our other SP releases. Is this standard?

Reply:
iirc we had a case where the customer wanted to get the code base of the image.

Reply:
Currently the action I envisioned does not create a new release in the SP rather open a PR in a completely separate repo. 
The one you are suggesting here handles a separate task in the release process (also important because that is still a manual) that could definitely be something to add in future developments section once I conquer the first obstacle. IMO your idea is much easier to implement then my original idea :)

For example, if a green bumped version bump:
1. grab content from changelog as the release description
2. Push the image and helm chart tarball
3. Create draft release. That way, there is still human involved to give the final OK

WDYT?